src/views/dialogs/SetMarketLeverageDialog.tsx

```diff
@@ -158,19 +168,40 @@ export const SetMarketLeverageDialog = ({
         throw new Error(stringGetter({ key: STRING_KEYS.INVALID_LEVERAGE_VALUE }));
       }
 
-      // TODO: This is currently a dummy transaction that just saves to local state.
-      // When this becomes a real chain transaction, it should return an OperationResult
-      // and the error handling pattern below will work correctly.
-      const result = await saveMarketLeverage({
-        dispatch,
-        marketId,
-        leverage: leverageBN.toNumber(),
-      });
+      if (clobPairId === undefined) {
+        throw new Error("clobPairId doesn't exist");
+      }
+
+      if (marketData === undefined) {
+        throw new Error('market data does not exist');
+      }
+
+      if (dydxAddress === undefined) {
+        throw new Error('dydx address does not exist');
+      }
 
-      if (isOperationSuccess(result)) {
+      const leverage = leverageBN.toNumber();
+
+      const subaccountNumber = currentPosition?.subaccountNumber;
+      if (subaccountNumber) {
```

Comment:
if subaccountNumber is 0 this will go to the else block. try `subaccountNumber != null`